dotnet/src/SemanticKernel.Abstractions/Events/PromptRenderingEventArgs.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.SemanticKernel.AI;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.Events;
+
+/// <summary>
+/// Event arguments available to the Kernel.PromptRendering event.
+/// </summary>
+public class PromptRenderingEventArgs : SKEventArgs
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PromptRenderingEventArgs"/> class.
+    /// </summary>
+    /// <param name="metadata">Function metadata</param>
+    /// <param name="context">Context related to the event</param>
+    /// <param name="requestSettings">request settings used by the AI service</param>
+    public PromptRenderingEventArgs(SKFunctionMetadata metadata, SKContext context, AIRequestSettings? requestSettings) : base(metadata, context)
```

Comment:
Is there a reason for all these events that we pass the metadata rather than the function itself?